File: Cmdlets/Cmdlets/GetCultureInfoBranchCommand.vb
<System.Management.Automation.Cmdlet("Get", "CultureInfoBranch")> _
Public Class GetCultureInfoBranchCommand
    Inherits System.Management.Automation.Cmdlet

    <System.Management.Automation.Parameter(Mandatory:=True)> _
    Public Property Name As String

    Protected Overrides Sub BeginProcessing()

        Dim cultureBranch As New System.Collections.Generic.List(Of System.Globalization.CultureInfo)

        Dim culture As New System.Globalization.CultureInfo(Me.Name)

        cultureBranch.Add(culture)

        AddParent(culture, cultureBranch)

        Me.WriteObject(cultureBranch, True)
    End Sub

    Private Sub AddParent(culture As System.Globalization.CultureInfo, cultureBranch As System.Collections.Generic.List(Of System.Globalization.CultureInfo))

        cultureBranch.Add(culture.Parent)

        If culture.Parent IsNot Nothing AndAlso Not culture.Parent.Equals(System.Globalization.CultureInfo.InvariantCulture) Then
            AddParent(culture.Parent, cultureBranch)
        End If
    End Sub

End Class

File: Cmdlets.Core/Commands/GetCultureInfoBranchCommand.cs
namespace TIKSN.Cmdlets.Core.Commands
{
    [System.Management.Automation.Cmdlet("Get", "CultureInfoBranch")]
    public class GetCultureInfoBranchCommand : System.Management.Automation.Cmdlet
    {
        [System.Management.Automation.Parameter(Mandatory = true)]
        public string Name { get; set; }

        protected override void BeginProcessing()
        {
            System.Collections.Generic.List<System.Globalization.CultureInfo> cultureBranch = new System.Collections.Generic.List<System.Globalization.CultureInfo>();

            System.Globalization.CultureInfo culture = new System.Globalization.CultureInfo(this.Name);

            cultureBranch.Add(culture);

            AddParent(culture, cultureBranch);

            this.WriteObject(cultureBranch, true);
        }

        private void AddParent(System.Globalization.CultureInfo culture, System.Collections.Generic.List<System.Globalization.CultureInfo> cultureBranch)
        {
            cultureBranch.Add(culture.Parent);

            if (culture.Parent != null && !culture.Parent.Equals(System.Globalization.CultureInfo.InvariantCulture))
            {
                AddParent(culture.Parent, cultureBranch);
            }
        }
    }
}